packages/gcf-utils/test/test-helpers.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import assert from 'assert';
+
+export interface LogLine {
+  msg: string;
+  level: number;
+  [index: string]: string | number;
+}
+
+/**
+ * Asserts the correctness of the provided log entries based on params given
+ * @param logs logged lines
+ * @param expectedLineCount expected number of log lines
+ * @param expectedMessages expected messages in each line (in order of log lines)
+ * @param expectedProperties expected properties and values in each line (in order of log lines)
+ * @param expectedLogLevel expected log level for all lines
+ */
+export function validateLogs(
+  logs: LogLine[],
+  expectedLineCount?: number,
+  expectedMessages?: string[],
+  expectedProperties?: Array<{[idx: string]: string}>,
+  expectedLogLevel?: number
+): void {
+  if (expectedLineCount) {
+    assert.equal(
+      logs.length,
+      expectedLineCount,
+      `expected exactly ${expectedLineCount} line(s) to be logged`
+    );
+  }
+  if (expectedMessages) {
+    for (let i = 0; i < expectedMessages.length; i++) {
```

Comment:
Is is possible that we log in separate async calls so that the order is non-deterministic? Do we want to assert on the exact order?

Reply:
In this case we do actually want to assert on the order because we're testing a synchronous logger (we pass `{sync:true}` to pino) so it should be flushing after each log statement.